view/range/index.ts

```diff
@@ -104,7 +107,7 @@ onPaint({
   ch(value, showP3, showRec2020) {
     let color = valueToColor(value)
     let c = color.c
-    let h = color.h ?? 0
+    let h = color.h ?? 0 // eslint-disable-line @typescript-eslint/no-unnecessary-condition
```

Comment:
`culori` could return LCH color without `h`.

For instance, `lch('black')` generated `{ mode: 'lch', l: 0, c: 0 }`

Reply:
Didn't notice that. I will recheck the types and add tests in the culori lib.

Reply:
`parseLab('lab(none none none / none)')` will return `{ mode: 'lab' }`.

![Screenshot from 2022-05-15 13-24-50](https://user-images.githubusercontent.com/102924398/168470329-615be24b-a719-41d6-95a0-fe8583eb6ecf.png)

So all props of Lab color type should be marked as optional.

![Screenshot from 2022-05-15 13-21-23](https://user-images.githubusercontent.com/102924398/168470182-daab7d69-2602-40ed-8b74-7b77b90a6ab3.png)

Reply:
Let’s avoid this case. I can’t image real use case (in contrast with `lab('black')`)

Reply:
An Rgb color object also can have a "missing" channel

![Screenshot from 2022-05-15 13-41-40](https://user-images.githubusercontent.com/102924398/168470979-4aabc8fa-1e59-489b-bc42-a53b18a3063f.png)

From my perspective the types should be honest. But I think it will make the types of the culori lib unusable in TS projects where `strictNullChecks` set to `true`.

I think it is better to introduce `parse` function which should return the color object where all properties are present, and where `"none"` keyword maps to a default value.

Reply:
> I think it is better to introduce parse function which should return the color object where all properties are present, and where "none" keyword maps to a default value.

Our current PR is about types. It is better to not change the current logic and open another issue/PR.

> What do you think?

Maybe making all keys optional is also a way since CSS Spec allows to use `none`.

Reply:
What if we will add `Rgb`/`Lch` (with all properties as number) and `PartialRgb`/`PartialLch` (where all property can be a number or `undefined`).

In types we will define multiple definitions for each functions: 

```ts
interface LchFunction {
  (color: string | PartialLch | PartialRgb): PartialLch
  (color: Lch | Rgb): Lch
} 

export const lch: LchFunction
```

Reply:
I meant the _new_ parsing function. But, you are right. This may require a lot of changes in the code base of the culori lib, which I wouldn't want to do right now.
So, yes, let's deal with the 'none' keyword in another PR.